fmralign/alignment_methods.py

```diff
@@ -465,3 +467,239 @@ def fit(self, X, Y):
+        - alignment_method (str): The alignment method to use. Can be either "searchlight" or "parcelation", Default is "searchlight".
+        - n_components (int): The number of latent dimensions to use in the shared stimulus information matrix. Default is None.
+        - n_jobs (int): The number of parallel jobs to run. Default is -1.
+
+        Returns:
+        --------
+        None
+        """
+
+        self.n_s = None
+        self.n_t = None
+        self.n_v = None
+        self.labels = None
+        self.alphas = None
+        self.alignment_method = alignment_method
+        if alignment_method == "parcelation":
+            self.parcels = None
+
+        elif (
+            alignment_method == "searchlight"
+            or alignment_method == "ensemble_searchlight"
+        ):
+            self.searchlights = None
+            self.distances = None
+            self.radius = None
+
+        self.tuning_data = []
+        self.denoised_signal = []
+        self.decomp_method = decomp_method
+        self.template = template
+        self.n_components = n_components
+        self.n_jobs = n_jobs
+
+    #######################################################################################
+    # Computing decomposition
+
+    @staticmethod
+    def _tuning_estimator(shared_response, target):
+        """
+        Estimate the tuning weights for individualized neural tuning.
+
+        Parameters:
+        --------
+            - shared_response (array-like):
+                The shared response matrix.
+            - target (array-like):
+                The target matrix.
+            - latent_dim (int, optional):
+                The number of latent dimensions. Defaults to None.
+
+        Returns:
+        --------
+            array-like: The estimated tuning weights.
+
+        """
+        if shared_response.shape[1] < shared_response.shape[0]:
+            return (np.linalg.pinv(shared_response)).dot(target)
+        return np.linalg.inv(shared_response).dot(target)
+
+    @staticmethod
+    def _stimulus_estimator(full_signal, n_t, n_s, latent_dim=None):
+        """
+        Estimates the stimulus response using the given parameters.
+
+        Args:
+            full_signal (np.ndarray): The full signal data.
+            n_t (int): The number of time points.
+            n_s (int): The number of stimuli.
+            latent_dim (int, optional): The number of latent dimensions. Defaults to None.
+
+        Returns:
+            stimulus (np.ndarray): The stimulus response of shape (n_t, latent_dim) or (n_t, n_t).
+        """
+        U = svd_pca(full_signal)
+        if latent_dim is not None and latent_dim < n_t:
+            U = U[:, :latent_dim]
+
+        stimulus = np.sqrt(n_s) * U
+        stimulus = stimulus.astype(np.float32)
+        return stimulus
+
+    @staticmethod
+    def _reconstruct_signal(shared_response, individual_tuning):
```

Comment:
This uses nothing from self, so this should be a function (or it is not even needed ?)